packages/track-event-mixin/track-event-gesture.ts

```diff
@@ -0,0 +1,308 @@
+        if (!hasLeftMouseButton(e)) {
+          // always fire "end"
+          self.info.state = 'end';
+          unTrackDocument(self.info);
+        }
+        /* istanbul ignore else */
+        if (target) {
+          trackFire(self.info, target, e);
+        }
+        self.info.started = true;
+      }
+    };
+    const upFn = function upFn(e: MouseEvent) {
+      if (self.info.started) {
+        moveFn(e);
+      }
+      // remove the temporary listeners
+      unTrackDocument(self.info);
+    };
+    // add temporary document listeners
+    trackDocument(this.info, moveFn, upFn);
+    this.info.x = event.clientX;
+    this.info.y = event.clientY;
+  },
+
+  touchstart(event: TouchEvent) {
+    const touch = event.changedTouches[0];
+    this.info.x = touch.clientX;
+    this.info.y = touch.clientY;
+  },
+
+  touchmove(event: TouchEvent) {
+    const target = event.composedPath()[0];
+    const touch = event.changedTouches[0];
+    const x = touch.clientX;
+    const y = touch.clientY;
+    if (trackHasMovedEnough(this.info, x, y)) {
+      this.info.addMove({ x, y });
+      trackFire(this.info, target, touch);
+      this.info.state = 'track';
+      this.info.started = true;
+    }
+  },
+
+  touchend(event: TouchEvent) {
+    const target = event.composedPath()[0];
+    const touch = event.changedTouches[0];
+    // only end if track was started and not aborted
+    if (this.info.started) {
+      // reset started state on up
+      this.info.state = 'end';
+      this.info.addMove({ x: touch.clientX, y: touch.clientY });
+      trackFire(this.info, target, touch);
+    }
+  }
+};
+
+const handleNative = (ev: MouseEvent | TouchEvent) => {
+  const { type } = ev;
+
+  if (type.slice(0, 5) === 'touch') {
+    const event = ev as TouchEvent;
+    const touch = event.changedTouches[0];
+    if (type === 'touchstart') {
+      // only handle the first finger
+      if (event.touches.length === 1) {
+        TOUCH_STATE.id = touch.identifier;
+      }
+    }
+    if (TOUCH_STATE.id !== touch.identifier) {
+      return;
+    }
+  }
+
+  if (type === 'mousedown' || type === 'touchstart') {
+    TrackGesture.reset();
+  }
+
+  switch (type) {
+    case 'mousedown':
+      TrackGesture.mousedown(ev as MouseEvent);
+      break;
+    case 'touchstart':
+      TrackGesture.touchstart(ev as TouchEvent);
+      break;
+    case 'touchmove':
+      TrackGesture.touchmove(ev as TouchEvent);
+      break;
+    case 'touchend':
+      TrackGesture.touchend(ev as TouchEvent);
+      break;
+    /* istanbul ignore next */
+    default:
+  }
+};
+
+/**
+ * Add event listeners for handling track.
+ */
+export function addTrackListeners(node: HTMLElement) {
+  const { deps } = TrackGesture;
+  for (let i = 0, dep; i < deps.length; i += 1) {
+    dep = deps[i];
+    // don't add mouse handlers on iOS because they cause gray selection overlays
+    /* istanbul ignore else */
+    if (Boolean(IS_TOUCH_ONLY && isMouseEvent(dep)) === false) {
```

Comment:
Comment doesn't seem accurate as it looks like this applies also to Android, not just iOS (based on definition of `IS_TOUCH_ONLY`)?

Reply:
Updated to use `isIOS`. We can test manually with Android later and update if needed.